File: app/src/main/java/com/app/playassetdeliverydemo/ui/VideoPlayerActivity.java
package com.app.playassetdeliverydemo.ui;

import android.annotation.SuppressLint;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;
import androidx.media3.common.MediaItem;
import androidx.media3.exoplayer.ExoPlayer;
import androidx.media3.ui.PlayerView;

import com.app.playassetdeliverydemo.R;

public class VideoPlayerActivity extends AppCompatActivity {

    private PlayerView playerView;
    private ExoPlayer player;
    private Uri uri;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_video_player);
        initViews();
    }

    /**
     * This method is used to initialize views
     */
    private void initViews() {
        playerView = findViewById(R.id.playerView);
        uri = getIntent().getData();
    }

    /**
     * lifecycle method to initialize ExoPlayer
     */
    @Override
    public void onStart() {
        super.onStart();
        if (Build.VERSION.SDK_INT >= 24) {
            initializePlayer();
        }
    }

    /**
     * lifecycle method to initialize ExoPlayer
     */
    @Override
    public void onResume() {
        super.onResume();
        hideSystemUi();
        if ((Build.VERSION.SDK_INT < 24 || player == null)) {
            initializePlayer();
        }
    }

    /**
     * lifecycle method to release ExoPlayer
     */
    @Override
    public void onPause() {
        super.onPause();
        if (Build.VERSION.SDK_INT < 24) {
            releasePlayer();
        }
    }

    /**
     * lifecycle method to release ExoPlayer
     */
    @Override
    public void onStop() {
        super.onStop();
        if (Build.VERSION.SDK_INT >= 24) {
            releasePlayer();
        }
    }

    /**
     * Method to initialize ExoPlayer
     */
    private void initializePlayer() {
        if (player == null) {
            player = new ExoPlayer.Builder(this).build();
            playerView.setPlayer(player);
            MediaItem mediaItem = MediaItem.fromUri(uri);
            player.setMediaItem(mediaItem);
            player.prepare();
            player.setPlayWhenReady(true);
        }
    }

    /**
     * Method to release ExoPlayer
     */
    private void releasePlayer() {
        if (player != null) {
            player.release();
            player = null;
        }
    }

    /**
     * Method to hide System UI & show ExoPlayer in FullScreen
     */
    @SuppressLint("InlinedApi")
    private void hideSystemUi() {
        playerView.setSystemUiVisibility(View.SYSTEM_UI_FLAG_LOW_PROFILE
                | View.SYSTEM_UI_FLAG_FULLSCREEN
                | View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY
                | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION);
    }
}

File: app/src/main/java/com/app/playassetdeliverydemo/ui/VideoPlayerActivity.kt
package com.app.playassetdeliverydemo.ui

import android.annotation.SuppressLint
import android.net.Uri
import android.os.Build
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat
import androidx.media3.common.MediaItem
import androidx.media3.exoplayer.ExoPlayer
import com.app.playassetdeliverydemo.databinding.ActivityVideoPlayerBinding

class VideoPlayerActivity : AppCompatActivity() {
    private lateinit var binding: ActivityVideoPlayerBinding
    private var player: ExoPlayer? = null
    private var uri: Uri? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityVideoPlayerBinding.inflate(layoutInflater)
        setContentView(binding.root)
        initViews()
    }

    /**
     * This method is used to initialize views
     */
    private fun initViews() {
        uri = intent.data
    }

    /**
     * lifecycle method to initialize ExoPlayer
     */
    public override fun onStart() {
        super.onStart()
        if (Build.VERSION.SDK_INT >= 24) {
            initializePlayer()
        }
    }

    /**
     * lifecycle method to initialize ExoPlayer
     */
    public override fun onResume() {
        super.onResume()
        hideSystemUi()
        if (Build.VERSION.SDK_INT < 24 || player == null) {
            initializePlayer()
        }
    }

    /**
     * lifecycle method to release ExoPlayer
     */
    public override fun onPause() {
        super.onPause()
        if (Build.VERSION.SDK_INT < 24) {
            releasePlayer()
        }
    }

    /**
     * lifecycle method to release ExoPlayer
     */
    public override fun onStop() {
        super.onStop()
        if (Build.VERSION.SDK_INT >= 24) {
            releasePlayer()
        }
    }

    /**
     * Method to initialize ExoPlayer
     */
    private fun initializePlayer() {
        if (player == null) {
            player = ExoPlayer.Builder(this@VideoPlayerActivity).build()
            binding.playerView.player = player
            val mediaItem = MediaItem.fromUri(
                uri ?: Uri.parse("")
            )
            player?.setMediaItem(mediaItem)
            player?.prepare()
            player?.playWhenReady = true
        }
    }

    /**
     * Method to release ExoPlayer
     */
    private fun releasePlayer() {
        if (player != null) {
            player?.release()
            player = null
        }
    }

    /**
     * Method to hide System UI & show ExoPlayer in FullScreen
     */
    @SuppressLint("InlinedApi")
    private fun hideSystemUi() {
        WindowCompat.setDecorFitsSystemWindows(window, false)
        WindowInsetsControllerCompat(window, binding.playerView).let { controller ->
            controller.hide(WindowInsetsCompat.Type.systemBars())
            controller.systemBarsBehavior =
                WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        }
    }
}
